packages/discovery-provider/src/challenges/play_count_milestone_challenge_base.py

```diff
@@ -38,7 +37,7 @@ def _get_user_play_count_2025(self, session: Session, user_id: int) -> int:
                     Track.is_delete == False,
                 ),
             )
-            .filter(and_(Play.created_at >= start_date, Play.created_at < end_date))
+            .filter(Play.created_at >= start_date)
```

Comment:
i'm worried about this being slow. it is a _very very_ hot path. like will run every time a single play event happens.

this query would be a lot faster if we ran it against the `aggregate_monthly_plays` table.

```
select count("plays"."id") from "plays"
join "tracks"
on "tracks"."track_id" = "plays"."play_item_id"
where "tracks"."owner_id" = 1
```
>> Ran for > 1 minute. No results on prod data.

```
select sum("count") from "aggregate_monthly_plays"
where "play_item_id" in (select "track_id" from "tracks" where "owner_id" = 1)
and "timestamp" > '2025-01-01'
```
> 300ms

Reply:
agree - one downside tho is that if `aggregate_plays` is behind on one node for some reason, we might get inconsistencies between DNs.
but that's probably the tradeoff we have to live with.. since many plays should be coming in eventually they should end up in the same state.